hypothesis-python/src/hypothesis/strategies/_internal/lazy.py

```diff
@@ -61,41 +63,49 @@ def unwrap_strategies(s):
         assert unwrap_depth >= 0
 
 
-class LazyStrategy(SearchStrategy):
+class LazyStrategy(SearchStrategy[Ex]):
     """A strategy which is defined purely by conversion to and from another
     strategy.
 
     Its parameter and distribution come from that other strategy.
     """
 
-    def __init__(self, function, args, kwargs, *, transforms=(), force_repr=None):
+    def __init__(
+        self,
+        function: Callable[..., SearchStrategy[Ex]],
+        args: Sequence[object],
+        kwargs: dict[str, object],
+        *,
+        transforms: tuple[tuple[str, Callable[..., Any]], ...] = (),
```

Comment:
It seems to me that we never actually pass this argument, so let's delete it and the fragments referencing it below.  (I think it was originally intended to be like the map/filter handling on SampledFromStrategy?)

Reply:
we do actually pass it in __with_transform, but it's obfuscated by `type(self)(...)`